### PR TITLE
Fix removal of status sending the original status to mentioned users instead of delete Salmon

### PR DESCRIPTION
**Pull request by Gargron**

### CODE REVIEW

File: spec/services/remove_status_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe RemoveStatusService do
+  subject { RemoveStatusService.new }
+
+  let!(:alice)  { Fabricate(:account, username: 'alice') }
```
Comment (akihikodaki) [inline spec/services/remove_status_service_spec.rb]: I do not think `alice` and `jeff` should have fixed usernames. Why not make use of Fabrication and Faker?

**Review by akihikodaki — DISMISSED** [review]

File: spec/services/remove_status_service_spec.rb (hunk 1 above)
Comment (Gargron) [inline spec/services/remove_status_service_spec.rb]: They are recurring characters in my tests

Comment (akihikodaki) [inline spec/services/remove_status_service_spec.rb]: That is true. But we have already introduced Faker as well. We should choose to fix usernames with those strings or use Faker, depending on the situations.
In this case I think Faker is better for `alice` and `jeff`, and a fixed username is better for `bob`.

**Comment by Gargron** [comment]
@akihikodaki Okay, fair enough. Done

**Comment by Gargron** [comment]
@akihikodaki

**Review by beatrix-bitrot — APPROVED** [review]
requested changes were implemented~

**Comment by unarist** [comment]
@Gargron This seems to break removing reblogs:

```
sidekiq_1    | 2017-06-12T00:44:19.280Z 1 TID-gpzcf49g8 WARN: NoMethodError: undefined method `nil?' for <Redis::Future [:zscore, "feed:home:7", 129]>:Redis::Future
sidekiq_1    | 2017-06-12T00:44:19.280Z 1 TID-gpzcf49g8 WARN: /mastodon/app/services/remove_status_service.rb:63:in `unpush'
sidekiq_1    | /mastodon/app/services/remove_status_service.rb:38:in `block (2 levels) in remove_from_followers'
sidekiq_1    | /usr/local/bundle/gems/activerecord-5.1.1/lib/active_record/relation/batches.rb:63:in `block (2 levels) in find_each'
sidekiq_1    | /usr/local/bundle/gems/activerecord-5.1.1/lib/active_record/relation/batches.rb:63:in `each'
sidekiq_1    | /usr/local/bundle/gems/activerecord-5.1.1/lib/active_record/relation/batches.rb:63:in `block in find_each'
sidekiq_1    | /usr/local/bundle/gems/activerecord-5.1.1/lib/active_record/relation/batches.rb:129:in `block in find_in_batches'
sidekiq_1    | /usr/local/bundle/gems/activerecord-5.1.1/lib/active_record/relation/batches.rb:230:in `block in in_batches'
sidekiq_1    | /usr/local/bundle/gems/activerecord-5.1.1/lib/active_record/relation/batches.rb:214:in `loop'
sidekiq_1    | /usr/local/bundle/gems/activerecord-5.1.1/lib/active_record/relation/batches.rb:214:in `in_batches'
sidekiq_1    | /usr/local/bundle/gems/activerecord-5.1.1/lib/active_record/relation/batches.rb:128:in `find_in_batches'
sidekiq_1    | /usr/local/bundle/gems/activerecord-5.1.1/lib/active_record/relation/batches.rb:62:in `find_each'
sidekiq_1    | /mastodon/app/services/remove_status_service.rb:37:in `block in remove_from_followers'
sidekiq_1    | /usr/local/bundle/gems/redis-3.3.3/lib/redis.rb:2257:in `block in pipelined'
sidekiq_1    | /usr/local/bundle/gems/redis-3.3.3/lib/redis.rb:58:in `block in synchronize'
sidekiq_1    | /usr/local/lib/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
sidekiq_1    | /usr/local/bundle/gems/redis-3.3.3/lib/redis.rb:58:in `synchronize'
sidekiq_1    | /usr/local/bundle/gems/redis-3.3.3/lib/redis.rb:2254:in `pipelined'
sidekiq_1    | /mastodon/app/services/remove_status_service.rb:36:in `remove_from_followers'
sidekiq_1    | /mastodon/app/services/remove_status_service.rb:16:in `call'
sidekiq_1    | /mastodon/app/workers/removal_worker.rb:7:in `perform'
...
```

Reblog deletion checks existence in the target timeline by `redis.zscore()`, but it doesn't return actual value because it's in the pipeline now.

**Comment by Gargron** [comment]
@unarist Ah... Using pipeline was not worth that.. I will fix

**Pull request merged**